plotly/plotly/plotly.py

```diff
@@ -1454,6 +1453,129 @@ def _send_to_plotly(figure, **plot_options):
     return r
 
 
+def create_animations(fig, kwargs, payload):
+    """
+    Makes a post to GRIDS and PLOTS if frames is in the figure.
+
+    This function bypasses the current '/clientresp' route  of making a POST
+    request to return back a url. Instead, the V2 REST API is used if frames
+    is part of the figure's keys. Currently, 'error', 'message' and 'warning'
+    are hard codded to the empty string.
+    """
+    url_v2_plot = "https://api.plot.ly/v2/plots"
+    url_v2_grid = "https://api.plot.ly/v2/grids"
+    auth = HTTPBasicAuth(str(payload['un']), str(payload['key']))
+    headers = {'Plotly-Client-Platform': 'python'}
```

Comment:
I think the `_api_v2` thing handles headers as well: https://github.com/plotly/plotly.py/blob/master/plotly/plotly/plotly.py#L674